backend/dataall/core/stacks/services/stack_service.py

```diff
@@ -99,3 +129,77 @@ def delete_stack(target_uri, accountid, cdk_role_arn, region):
 
         Worker.queue(context.db_engine, [task.taskUri])
         return True
+
+    @staticmethod
+    def get_stack_by_uri(stack_uri):
+        with get_context().db_engine.scoped_session() as session:
+            return StackRepository.get_stack_by_uri(session, stack_uri)
+
```

Comment:
I think this method, `get_environmental_stack_by_uri` should go back to the environment, and should be called `get_environment_stack_by_uri`